source/model.js

```diff
@@ -37,52 +50,97 @@ let singletonField = (function(){
     };
 
     return {
-        getInstance: function(){
+        getInstance: function(rows, columns){
```

Comment:
Singletone's `getInstance` can't contain parameters. There should be a better approach. I have a suggestion about this...